python/GangaLSST/Lib/Im3ShapeApp/Im3ShapeApp.py

```diff
@@ -0,0 +1,114 @@
+##########################################################################
+# Ganga Project. http://cern.ch/ganga
+##########################################################################
+
+from Ganga.GPIDev.Adapters.IGangaFile import IGangaFile
+from Ganga.GPIDev.Adapters.IPrepareApp import IPrepareApp
+from Ganga.GPIDev.Adapters.IRuntimeHandler import IRuntimeHandler
+from Ganga.GPIDev.Schema import Schema, Version, SimpleItem, ComponentItem, GangaFileItem
+
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File import File, ShareDir
+from Ganga.Core import ApplicationConfigurationError, ApplicationPrepareError
+
+from Ganga.Utility.logging import getLogger
+
+from Ganga.GPIDev.Base.Proxy import getName, isType, stripProxy
+
+import os
+import shutil
+from Ganga.Utility.files import expandfilename
+
+from GangaDirac.Lib.Files import DiracFile
+from Ganga.GPIDev.Lib.File import LocalFile
+
+logger = getLogger()
+
+class Im3ShapeApp(IPrepareApp):
+
+    """
+    This App is to store the configuration of the Im3Shape app which is to be run according to a given set of configs:
+
+    i.e.
+    ./run_dir/exe_name <someData> ini_location catalog <someOutput> rank size
+
+    e.g.
+    ./run_dir/run-im3shape someData.fz ini_file.ini all someData.fz.0.20 0 20
+
+    The input and output file names are configured in the RTHandler based upon the inputdata given to a particular job.
+
+    The im3_location is the path to a .tgz file (or some file which can be extracted by the RTHandler) which gives the im3shape-grid (run_dir) folder containing the run-im3shape app (exe-name) on the WN
+    """
+    _schema = Schema(Version(1, 0), {
+        ## Required to configure Im3ShapeApp
+        'im3_location': GangaFileItem(defvalue=None, doc="Location of the Im3Shape program tarball"),
+        'exe_name': SimpleItem(defvalue='run-im3shape', doc="Name of the im3shape binary"),
+        'ini_location': GangaFileItem(defvalue=None, doc=".ini file used to configure Im3Shape"),
+        'blacklist': GangaFileItem(defvalue=None, doc="Blacklist file for running Im3Shape"),
+        'rank': SimpleItem(defvalue=0, doc="Rank in the split of the tile from splitting"),
+        'size': SimpleItem(defvalue=200, doc="Size of the splitting of the tile from splitting"),
+        'catalog': SimpleItem(defvalue='all', types=[str], doc="Catalog which is used to describe what is processed"),
+        'run_dir': SimpleItem(defvalue='im3shape-grid', types=[str], doc="Directory on the WN where the binary is"),
+        ## Below is needed for prepared state stuff
+        'is_prepared': SimpleItem(defvalue=None, strict_sequence=0, visitable=1, copyable=1, hidden=0, typelist=[None, bool, ShareDir], protected=0, comparable=1, doc='Location of shared resources. Presence of this attribute implies the application has been prepared.'),
```

Comment:
Since we're in control of this app and we can decide its semantics, I feel this should be set to `typelist=[None, ShareDir]`. Especially since you do things like `self.is_prepared.name` below.